packages/hoppscotch-app/components/collections/graphql/AddRequest.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <SmartModal
+    v-if="show"
+    dialog
+    :title="$t('request.new')"
+    @close="$emit('hide-modal')"
+  >
+    <template #body>
+      <div class="flex flex-col px-2">
+        <input
+          id="selectLabelGqlAddRequest"
+          v-model="name"
+          v-focus
+          class="input floating-input"
+          placeholder=" "
+          type="text"
+          autocomplete="off"
+          @keyup.enter="addRequest"
+        />
+        <label for="selectLabelGqlAddRequest">
+          {{ $t("action.label") }}
+        </label>
+      </div>
+    </template>
+    <template #footer>
+      <span>
+        <ButtonPrimary :label="$t('action.save')" @click.native="addRequest" />
+        <ButtonSecondary
+          :label="$t('action.cancel')"
+          @click.native="hideModal"
+        />
+      </span>
+    </template>
+  </SmartModal>
+</template>
+
+<script setup lang="ts">
+import { ref, watch } from "@nuxtjs/composition-api"
+import { useI18n, useToast } from "~/helpers/utils/composables"
+import { useGQLRequestName } from "~/newstore/GQLSession"
+
+const toast = useToast()
+const t = useI18n()
+
+const props = defineProps<{
+  show: boolean
+  folderPath?: string
+}>()
+
+const emit = defineEmits<{
+  (e: "hide-modal"): void
+  (e: "add-request", v: object): void
+}>()
+
+const name = ref("")
+
+watch(props, (v) => {
```

Comment:
```suggestion
watch(() => props.show, (show) => {) // 'show' will have the value of the show prop.
```

Reply:
What is the performance difference if we use `onBeforeUpdate` instead of `watch`?